docs: add documentation for influxdb scaler (#315)
kedacore/keda-docs: content/docs/2.1/scalers/influxdb.md

+++
title = "InfluxDB"
layout = "scaler"
availability = "v2.1+"
maintainer = "Community"
description = "Scale applications based on InfluxDB queries"
+++

### Trigger Specification

This specification describes the `InfluxDB` trigger that scales based on the results of a flux query.

```yaml
triggers:
  - type: influxdb
    metadata:
      serverURL: http://influxdb:8086
      organizationName: influx-org

>       organizationNameFromEnv: INFLUXDB_ORG_NAME

Reviewer: Can we comment this one out, or at least mention that you either have to choose between `organizationName` or `organizationNameFromEnv` and not both please?

Author: @tomkerkhove Yeah no problem. I was following the Redis List example here: https://keda.sh/docs/2.0/scalers/redis-lists/

Which is why I added both, but I will comment the `env` ones out.

Reviewer: Thanks! Will update those asap then to align as well.

      thresholdValue: '4'
      query: |
        from(bucket: "bucket-of-interest")
        |> range(start: -12h)
        |> filter(fn: (r) => r._measurement == "stat")

>       authToken: some-auth-token

Reviewer: Can we comment this one out, or at least mention that you either have to choose between `authToken` or `authTokenFromEnv` and not both please?

+++
title = "InfluxDB"
layout = "scaler"
availability = "v2.1+"
maintainer = "Community"
description = "Scale applications based on InfluxDB queries"
+++

### Trigger Specification

This specification describes the `InfluxDB` trigger that scales based on the results of a flux query.

```yaml
triggers:
  - type: influxdb
    metadata:
      serverURL: http://influxdb:8086
      organizationName: influx-org
      organizationNameFromEnv: INFLUXDB_ORG_NAME # Optional: You can use this instead of `organizationName` parameter. See details in "Parameter List" section
      thresholdValue: '4'
      query: |
        from(bucket: "bucket-of-interest")
        |> range(start: -12h)
        |> filter(fn: (r) => r._measurement == "stat")
      authToken: some-auth-token
      authTokenFromEnv: INFLUXDB_AUTH_TOKEN # Optional: You can use this instead of `authToken` parameter. See details in "Parameter List" section

>       metricName: influx-metric # Optional: This value will default to a masked version of the url if not set by the user (generated value would be `influxdb-influx-metric-influx-org`)

Reviewer 2: ```suggestion
      metricName: influx-metric # Optional: This value will default to a masked version of the url if not set by the user (generated value would be `influxdb-https---xxx-influx_org`)
```

Author: No this is actually correct, because the `metricName` is provided. The value you suggest is a default.

Author: I will add documentation for when the value is defaulting though

>       metricName: influx-metric # Optional: This value will default to a masked version of the url if not set by the user (generated value would be `influxdb-influx-metric-influx-org`). If this value isn't set the generated value would be (`influxdb-https---xxx-influx_org`)

Reviewer 2: ```suggestion
      metricName: influx-metric # Optional: This value will default to a masked version of the url and organization name if not set by the user (metrics name value would be then `influxdb-https---xxx-influx_org`).
```

I'd rephrase this a little bit, the previous version was a little bit confusing. imho.

```

**Parameter list:**

- `authToken` field holds the authentication token needed for the InfluxDB client to communicate with an associated server. 
- `authTokenFromEnv` defines the authorization token, similar to `authToken`, but reads it from an environment variable on the scale target.
- `organizationName` is the organization name needed for the client to locate all information contained in that [organization](https://docs.influxdata.com/influxdb/v2.0/organizations/) such as buckets, tasks, etc.
- `organizationNameFromEnv` defines the organization name, similar to `organizationName`, but reads it from an environment variable on the scale target.
- `serverURL` holds the url value of the InfluxDB server.
- `thresholdValue` is provided by the user. This value can vary from use case to use case depending on the data of interest, and is needed to trigger the scaling in/out depending on what value comes back from the query.
- `query` is the flux query that will yield the value for the scaler to compare the `thresholdValue` against.

> - `metricName` is an optional parameter that a user can give to provide some uniqueness of `ScaledObject` names. It's optional and will default in source code if not provided as a parameter.

Reviewer 2: Minor nit here, could you please use the similar description (and examples) for `metricName` that's being used in this PR: https://github.com/kedacore/keda-docs/pull/313

> - `metricName` is an optional parameter that a user can give to provide uniqueness of `ScaledObject` names. If not set KEDA will generate a unique name based on time. It is required that all metric names are unique, if using more than one trigger.

Reviewer 2: when reading this I have found out a problem, that I didn't spot in the original PR: 
https://github.com/kedacore/keda/blob/af2b36f593c231ed25c48a0ba432f6128f77cbf5/pkg/scalers/influxdb_scaler.go#L77

this is not ideal, the metric name should be immutable, ie. timestamp is not the right way. Could you please revert it and base in on organization name and/or serverURL (the same way it is done for postgresql scaler?) Thanks!

Reviewer 2: I am sorry I didn't find this when doing the review on the original PR

Author: @zroubalik No worries. So the serverURL and organization name is not enough to make it unique. I can take a look into how Postgres does it.

Reviewer 2: @ycabrer yeah, but we should do the best effort for that and we can expect that majority of ScaledObjects will use just one scaler. We are implementing a solution for check metric name uniqueness on general level: https://github.com/kedacore/keda/pull/1390

Reviewer 2: @yquansah any update on the metric name please?

Author: @zroubalik Hello I've been slammed with work over the past week i apologize! I will get a PR in by today or tomorrow 👍

Reviewer 2: @yquansah no worries and no rush, just checking the status :)

Author: @zroubalik https://github.com/kedacore/keda/pull/1429 Please let me know feedback on this then I can update this PR as well

Reviewer 2: @yquansah thanks, I merged the PR, could you please rephrase this section a bit?
